src/psykal/Makefile

```diff
@@ -10,15 +10,19 @@ all: tra_adv.exe
 
 # Location of the dl_timer code
 TIMER_DIR = ../shared/dl_timer
+INF_DIR = ../shared/dl_esm_inf/finite_difference
 
-FORT_FLAGS = ${F90FLAGS} -I${TIMER_DIR}/src
+# Append the locations of the various library modules to the
+# list of compiler flags
+FORT_FLAGS = ${F90FLAGS} -I${TIMER_DIR}/src -I${INF_DIR}/src
 
 KERN_OBJ = zind_kern.o zwxy_kern.o zslpxy_kern.o zslpxy_update_kern.o zwxy2_kern.o \
 	   mydomain_update_kern.o zwx_kern.o zslpx_kern.o zslpx_update_kern.o \
 	   zwx2_kern.o mydomain_kern.o
 
-tra_adv.exe: timer_lib ${KERN_OBJ} psy.o tra_adv_alg.o
-	${F90} -o $@ tra_adv_alg.o psy.o ${KERN_OBJ} ${TIMER_DIR}/dl_timer_lib.a
+tra_adv.exe: timer_lib inf_lib ${KERN_OBJ} psy.o tra_adv_alg.o
+	${F90} -o $@ tra_adv_alg.o psy.o ${KERN_OBJ} \
+        ${TIMER_DIR}/dl_timer_lib.a ${INF_DIR}/src/dl_esm_inf_fd.a
```

Comment:
Strictly speaking, the .a file is just an archive of object files and can therefore be simply included in the list of objects to link together.